crates/uv-pep508/src/marker/algebra.rs

```diff
@@ -181,6 +181,53 @@ impl InternerGuard<'_> {
                     }
                 }
             },
+            MarkerExpression::VersionInvertedTilde { key, specifier } => match key {
```

Comment:
All the other branches call `normalize_specifier`, this one probably should to. That might be the cause of the test failure (but I'm not sure why the non-normalized version would be comparing equal to the normalized).

Reply:
I found some pre-existing inconsistencies: https://github.com/astral-sh/uv/issues/14270

Reply:
We're now doing entirely different normalization